src/streamlink/plugins/mdstrm.py

```diff
@@ -0,0 +1,140 @@
+"""
+$description CDN hosting live content for various websites in the Americas.
+$url mdstrm.com
+$type live
+"""
+import logging
+import re
+from urllib.parse import urlparse
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+from streamlink.utils.url import update_scheme
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https://(?:www\.)?latina\.pe/tvenvivo"
+))
+@pluginmatcher(re.compile(
+    r"https://saltillo\.multimedios\.com/video"
+))
+@pluginmatcher(re.compile(
+    r"https://mdstrm\.com/live-stream/\w+"
+))
+class MDStrm(Plugin):
+    _re_ad = re.compile(r"parent\._dai_session\s*=\s*'([^']+)';")
+
+    def get_script_str(self, root, search_string, custom_pattern=None, custom_schema=None):
+        if custom_pattern:
+            pattern = custom_pattern
+        else:
+            pattern = fr"{search_string}\s*=\s*'([^']+)';"
+        _schema = validate.Schema(
+            validate.xml_xpath_string(
+                f".//script[@type='text/javascript'][contains(text(),'{search_string}')]/text()"),
+            validate.any(None, validate.all(
+                validate.transform(re.compile(pattern).search),
+                validate.any(None, validate.all(validate.get(1), str)))))
+        _string = validate.validate(_schema, root)
+        if not _string:
+            log.debug(f"Failed to find {search_string}")
+        if custom_schema:
+            try:
+                _string = validate.validate(custom_schema, _string)
+            except ValueError:
+                pass
+        return _string
+
+    def _get_streams(self):
+        p_netloc = urlparse(self.url).netloc
+        if p_netloc == "mdstrm.com":
+            url_iframe = self.url
+        else:
+            url_iframe = self.session.http.get(
+                url=self.url,
+                schema=validate.Schema(
+                    validate.parse_html(),
+                    validate.xml_xpath_string("normalize-space(.//iframe[contains(@src,'mdstrm.com')]/@src)")))
+            if not url_iframe:
+                return
+
+        url_iframe = update_scheme("https://", url_iframe, force=False)
+        log.debug(f"iframe={url_iframe}")
+        root = self.session.http.get(url_iframe,
+                                     schema=validate.Schema(validate.parse_html()))
+
+        schema = validate.Schema(validate.xml_xpath_string(".//div[@id='message']/text()"),
+                                 validate.any(None, str))
+        error_msg = validate.validate(schema, root)
+        if error_msg:
+            log.error(f"{error_msg}")
```

Comment:
Is this intentional that there's no `return` statement here? If the content is unavailable / geo-blocked, then it should stop here (I guess), because otherwise it'll log
```
[plugins.mdstrm][debug] Failed to find window.MDSTRM.OPTIONS
```

Reply:
it is intended,
i don't know if `".//div[@id='message']/text()"` is used for other messages